symbolic-base/test/Main.hs

```diff
@@ -1,46 +1,52 @@
-import           Tests.ByteString          (specByteString)
-import           Tests.Compiler            (specCompiler)
-import           Tests.EllipticCurve       (specEllipticCurve)
-import           Tests.FFA                 (specFFA)
-import           Tests.Field               (specField)
-import           Tests.GroebnerBasis       (specGroebner)
-import           Tests.Group               (specAdditiveGroup)
-import           Tests.Hash                (specHash)
-import           Tests.IVC                 (specIVC)
-import           Tests.List                (specList)
-import           Tests.NonInteractiveProof (specNonInteractiveProof)
-import           Tests.Pairing             (specPairing)
-import           Tests.Permutations        (specPermutations)
-import           Tests.Plonkup             (specPlonkup)
-import           Tests.ReedSolomon         (specReedSolomon)
-import           Tests.RSA                 (specRSA)
-import           Tests.SHA2                (specSHA2, specSHA2Natural)
-import           Tests.UInt                (specUInt)
-import           Tests.Univariate          (specUnivariate)
+import           Prelude                                 hiding (Bool, Fractional (..), Num (..), drop, length,
+                                                          replicate, take, (==))
+import           Tests.Base.Algebra.EllipticCurve        (specEllipticCurve)
+import           Tests.Base.Algebra.Field                (specField)
+import           Tests.Base.Algebra.GroebnerBasis        (specGroebner)
+import           Tests.Base.Algebra.Group                (specAdditiveGroup)
+import           Tests.Base.Algebra.Pairing              (specPairing)
+import           Tests.Base.Algebra.Permutations         (specPermutations)
+import           Tests.Base.Algebra.ReedSolomon          (specReedSolomon)
+import           Tests.Base.Algebra.Univariate           (specUnivariate)
+import           Tests.Base.Data.Binary                  (specBinary)
+import           Tests.Base.Protocol.IVC                 (specIVC)
+import           Tests.Base.Protocol.NonInteractiveProof (specNonInteractiveProof)
+import           Tests.Base.Protocol.Plonkup             (specPlonkup)
+import           Tests.Symbolic.Algorithm.Blake2b        (specBlake2b)
+import           Tests.Symbolic.Algorithm.RSA            (specRSA)
+import           Tests.Symbolic.Algorithm.SHA2           (specSHA2, specSHA2Natural)
+import           Tests.Symbolic.ArithmeticCircuit        (specArithmeticCircuit)
+import           Tests.Symbolic.Compiler                 (specCompiler)
+import           Tests.Symbolic.Data.ByteString          (specByteString)
+import           Tests.Symbolic.Data.FFA                 (specFFA)
+import           Tests.Symbolic.Data.Hash                (specHash)
+import           Tests.Symbolic.Data.List                (specList)
+import           Tests.Symbolic.Data.UInt                (specUInt)
 
 main :: IO ()
 main = do
```

Comment:
I guess it would also be more idiomatic to refactor `Tests.*` modules so that they would export [Spec](https://hackage.haskell.org/package/hspec-2.11.10/docs/Test-Hspec.html#t:Spec)s instead of `IO ()` actions and run them inside a single toplevel `hspec`.